private/bufpkg/bufplugin/bufpluginconfig/bufpluginconfig.go

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package bufpluginconfig defines the buf.plugin.yaml file.
+package bufpluginconfig
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/bufbuild/buf/private/bufpkg/bufplugin/bufpluginref"
+	"github.com/bufbuild/buf/private/pkg/encoding"
+	"github.com/bufbuild/buf/private/pkg/storage"
+)
+
+const (
+	// ExternalConfigFilePath is the default configuration file path for v1.
+	ExternalConfigFilePath = "buf.plugin.yaml"
+	// V1Version is the version string used to indicate the v1 version of the buf.plugin.yaml file.
+	V1Version = "v1"
+)
+
+var (
+	// AllConfigFilePaths are all acceptable config file paths without overrides.
+	//
+	// These are in the order we should check.
+	AllConfigFilePaths = []string{
+		ExternalConfigFilePath,
+	}
+)
+
+// Config is the plugin config.
+type Config struct {
+	// Name is the name of the plugin (e.g. 'buf.build/protocolbuffers/go').
+	Name bufpluginref.PluginIdentity
+	// Options is the set of options available to the plugin.
```

Comment:
What does it mean by options available to the plugin. Isn't this the default options that we pass to the plugin?

Reply:
Yea I can rephrase this comment a bit - they are the default options passed into the plugin.